src/test/java/org/kohsuke/github/junit/GitHubWireMockRule.java

```diff
@@ -235,6 +264,8 @@ protected void after() {
         recordSnapshot(this.codeloadServer(), "https://codeload.github.com", true);
 
         recordSnapshot(this.actionsUserContentServer(), "https://pipelines.actions.githubusercontent.com", true);
+
+        recordSnapshot(this.blobCoreWindowsNetServer(), "https://productionresults.blob.core.windows.net", true);
     }
 
     private void recordSnapshot(WireMockServer server, String target, boolean isRawServer) {
```

Comment:
If you look at the `recordSnapshot`, we add headers to the recording that are needed by tests.  

To test this feature, we need to verify that the Authorization header is present on the regular requests and that it is stripped from the redirected request.

To do that I think we need to an option to use/record/verify a placeholder `Authorization` header as part of a test.  You've done a ton of work here already.  If you'd like, I can do make this update.